client/service/src/builder.rs

```diff
@@ -877,10 +877,32 @@ where
 			.collect::<Vec<_>>(),
 	};
 
+	// crate transactions protocol and add it to the list of supported protocols of `network_params`
+	let transactions_handler_proto = sc_network_transactions::TransactionsHandlerPrototype::new(
+		protocol_id.clone(),
+		client
+			.block_hash(0u32.into())
+			.ok()
+			.flatten()
+			.expect("Genesis block exists; qed"),
+		config.chain_spec.fork_id(),
+	);
+	network_params
+		.network_config
+		.extra_sets
+		.insert(0, transactions_handler_proto.set_config());
```

Comment:
Just curious: how `extra_sets` are ordered? Why are we prepending? This seems to be error-prone...

Reply:
Good point. We should not need this anymore?

Reply:
Removed

Reply:
Did you may forgot to push the changes?

Reply:
I added it back to extra sets to the protocol is advertised. As we discussed, I tested the TX code with the default handshake and it worked so resolving this and the other comment mentioning this issue.